crypto/ex_data.c

```diff
@@ -287,8 +287,6 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         CRYPTOerr(CRYPTO_F_CRYPTO_DUP_EX_DATA, ERR_R_MALLOC_FAILURE);
         return 0;
     }
-    if (!CRYPTO_set_ex_data(to, mx - 1, NULL))
-        goto err;
```

Comment:
>in ssl_session_dup the ex_data is initialized to zero with

That's only in ssl_session_dup(). SSL, EC_KEY and BIO_dup_chain() do not do that. 
This change also updates the ssl_session_dup() to initialize ex_data to make it consistent with those other objects.